zookeeper-server/src/main/java/org/apache/zookeeper/server/quorum/LearnerHandler.java

```diff
@@ -155,12 +176,42 @@ public synchronized boolean check(long time) {
 
     private SyncLimitCheck syncLimitCheck = new SyncLimitCheck();
 
+    private static class MarkerQuorumPacket extends QuorumPacket {
+        long time;
+        MarkerQuorumPacket(long time) {
+            this.time = time;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(super.hashCode(), time);
```

Comment:
This is a private class used for tracking, won't change super fields, time is the only changing variable here, so only need to hash time here.

Reply:
done